portal/src/util/phone.ts

```diff
@@ -39,3 +39,11 @@ export function makePartialValue(
   const trimmed = trimCountryCallingCode(rawInputValue, countryCallingCode);
   return `+${countryCallingCode}${trimmed}`;
 }
+
+export function validatePhoneNumber(inp: string): string | null {
+  const lookLike = /^\s*\+[-0-9]+\s*$/.test(inp);
```

Comment:
Actually we should allow any spaces, hyphens in the middle. And this function should not be named `validate`. It is not doing validation. It is cleansing an input, and see if it looks like a phone number. If yes, then a cleansed phone number is returned.

To simplify, let's install `"libphonenumber-js": "1.10.53"` (this package is installed by authui already). and use `import { default as parsePhoneNumber } from "libphonenumber-js"`. Write a test to showcase the behavior of `parsePhoneNumber` is something we expect.

```
function parsePhoneNumber(input: string): string | null {
  const parsed = libphonenumberParsePhoneNumber(input);
  if (parsed == null) {
    return null;
  }
  return parsed.format("E.164");
}
```

```
parsePhoneNumber("+8522") => "+8522"
parsePhoneNumber("+852") => "+852"
parsePhoneNumber("") => null
parsePhoneNumber("+") => null
parsePhoneNumber("+1") => "+1"
```